File: SDKs/amazonaws.com/kinesisanalytics/2015-08-14/java/lib/src/main/java/openapisdk/models/shared/KinesisFirehoseInputDescription.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * KinesisFirehoseInputDescription
 *  Describes the Amazon Kinesis Firehose delivery stream that is configured as the streaming source in the application input configuration. 
**/
public class KinesisFirehoseInputDescription {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ResourceARN")
    public String resourceARN;
    public KinesisFirehoseInputDescription withResourceArn(String resourceARN) {
        this.resourceARN = resourceARN;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("RoleARN")
    public String roleARN;
    public KinesisFirehoseInputDescription withRoleArn(String roleARN) {
        this.roleARN = roleARN;
        return this;
    }
}

File: SDKs/amazonaws.com/kinesisanalyticsv2/2018-05-23/python/src/sdk/models/shared/kinesisfirehoseinputdescription.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class KinesisFirehoseInputDescription:
    r"""KinesisFirehoseInputDescription
    Describes the Amazon Kinesis Data Firehose delivery stream that is configured as the streaming source in the application input configuration. 
    """
    
    resource_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ResourceARN') }})
    role_arn: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('RoleARN') }})
    
